[FEAT/#215] 온보딩 이용약관 바텀시트 구현

angryPodo opened the pull request:
## Related issue 🛠
- closed #215 

## Work Description ✏️
- 바텀시트 구현
- 바텀시트 기본값 변경
- Url 중앙화
- 웹링크 함수화

## Screenshot 📸

https://github.com/user-attachments/assets/0b3f7119-58e0-464f-9049-7aff5f52bae9

## To Reviewers 📢
어쩌다보니 작업이 많아졌는데요
바텀시트 만들다 보니 Url이 필요해서 중앙화 시켰고
함수만드는게 귀찮아져서 코어로 뺏고...
의존성 이상하게 들어가있는것도 고치고.....
이것저것했습니다.

angryPodo committed:
[MOD/#215] Url 중앙화

angryPodo committed:
[MOD/#215] Url 중앙화

angryPodo committed:
[MOD/#215] 린트 수정

angryPodo committed:
[MOD/#215] 온보딩 이용 약관 동의 바텀시트 구현

angryPodo committed:
[MOD/#215] 온보딩 이용 약관 동의 바텀시트 구현

angryPodo committed:
[MOD/#215] 온보딩 이용 약관 동의 바텀시트 구현

nahy-512 commented on core/common/src/main/java/com/hilingual/core/common/constant/UrlConstant.kt:
```diff
@@ -0,0 +1,8 @@
+object UrlConstant {
+    const val PRIVACY_POLICY = "https://hilingual.notion.site/230829677ebf8104b52ce74c65c27607"
+    const val SERVICE_TERMS = "https://hilingual.notion.site/230829677ebf817a8091f49423cbbb11"
+    const val ONBOARDING_PRIVACY_POLICY = "https://hilingual.notion.site/230829677ebf8104b52ce74c65c27607"
```
p3
이거 개인정보 처리방침 앱 내에서 다 같은 거 쓰는 거 아닌가요??
온보딩 용을 분리하신 이유가 있는지 궁금합니다

angryPodo replied:
추후에 정책 관련해서 수정사항 생기면 수정 용이하게 하려고ㅎㅎ

nahy-512 replied:
어떤 식의 수정사항일지 더 자세한 설명 부탁드려도 될까요?

angryPodo replied:
저도 정책을 자세히 아는건 아니지만 개인정보처리방침과 서비스 이용약관 동의 고지 내용이 달라야 한다는 상황도 있어서요. 추후에 링크 하나 더 추가할때 변수선언해주고 호출부 바꿔주고 하는거보다 링크만 갈아끼우도록 한겁니다

Hyobeen-Park replied:
개인정보처리방침과 서비스 이용약관 각각 고지해야 하는거 맞습니다!! 근데 온보딩에서도 개인정보처리방침 / 서비스 이용약관 두 개 나눠서 보여주고 있는 것 같은데 온보딩용 링크가 필요한가요...? (그리고 애초에 앱 내에서 보여주는거랑 온보딩에서 보여주는게 달라질 수 있으면 안될 것 같음...)

angryPodo replied:
서비스 이용약관 동의도 프라이버시 폴리시라고 생각해서 저렇게 네이밍햇수

angryPodo replied:
> 근데 온보딩에서도 개인정보처리방침 / 서비스 이용약관 두 개 나눠서 보여주고 있는 것 같은데 온보딩용 링크가 필요한가요...? (그리고 애초에 앱 내에서 보여주는거랑 온보딩에서 보여주는게 달라질 수 있으면 안될 것 같음...)

이거 이해못했어

nahy-512 commented on presentation/onboarding/src/main/java/com/hilingual/presentation/onboarding/component/TermsBottomSheet.kt:
```diff
@@ -0,0 +1,189 @@
+    var marketingAgreed by remember { mutableStateOf(false) }
+
+    val allAgreed = serviceAgreed && privacyAgreed && marketingAgreed
+    val isStartButtonEnabled = serviceAgreed && privacyAgreed
```
p2
변수 자체도 많고, 조건 체크도 조금 하드 코딩이지 않나,, 하는 생각이 조금 드는데요.(약관이 추후 더 늘어날 수도 있을 것 같고) 약관 중 필수 약관을 따로 관리하는 건 별로일까요?
'약관 문구 + 링크(optional) + 필수/선택 여부'를 Term으로 묶어서 관리하는 방법은 어떻게 생각하시는지 궁금해요!

angryPodo replied:
좋은거 같아요. 재사용안할 컴포넌트라서 최대한 간단하게 짜려고 했던거라 약관 늘어나는건 생각 못했네요ㅎㅎ 고민해서 적용해보겠습니다

nahy-512 commented on presentation/onboarding/src/main/java/com/hilingual/presentation/onboarding/component/TermsBottomSheet.kt:
```diff
@@ -0,0 +1,189 @@
+            imageVector = ImageVector.vectorResource(R.drawable.ic_check_circle_28_and),
+            tint = if (isSelected) HilingualTheme.colors.black else Color.Unspecified,
+            modifier = Modifier.noRippleClickable(onClick = { onSelectedChange(!isSelected) }),
+            contentDescription = "$text 체크박스"
```
p2
저희 전에 컨벤션으로 contentDescription은 다 null로 두자고 얘기했던 걸로 기억하는데, 여기서 추가해주신 이유가 있으실까요?

angryPodo replied:
아뇨 잼민이 돌리다가 들어갔나봐요ㅎㅎ

nahy-512 commented on presentation/onboarding/src/main/java/com/hilingual/presentation/onboarding/component/TermsBottomSheet.kt:
```diff
@@ -0,0 +1,189 @@
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(horizontal = 16.dp)
+                    .padding(top = 32.dp, bottom = 16.dp)
```
p1
바텀패딩 버튼 밑 영역으로 이해했는데, 디자인 상으로 12.dp 아닌가요??

angryPodo replied:
분명 16이었는데..ㅋㅋ 수정하겠습니다

nahy-512 commented on presentation/onboarding/src/main/java/com/hilingual/presentation/onboarding/component/TermsBottomSheet.kt:
```diff
@@ -0,0 +1,189 @@
+                        Icon(
+                            imageVector = ImageVector.vectorResource(R.drawable.ic_check_circle_28_and),
+                            tint = if (allAgreed) HilingualTheme.colors.black else Color.Unspecified,
+                            contentDescription = "전체 동의 체크박스"
```
여기도 contentDescription 들어가있네여

angryPodo committed:
[MOD/#215] contentDescription null 적용, 패딩 조절

angryPodo committed:
[MODE/#215] Refactor TermsBottomSheet for better state management

This commit refactors the `TermsBottomSheet` composable to improve how terms and their agreement states are managed.

- Introduced `TermType` enum and `Term` data class to represent different types of terms and their properties.
- Replaced individual `mutableStateOf` for each term with a `persistentMapOf` to store agreement states, keyed by `Term` objects.
- Updated logic for `allAgreed`, `isStartButtonEnabled`, and `isMarketingAgreed` to derive from the new `agreements` state.
- Modified the rendering of `TermRow`s to iterate over the `agreements` map.

nahy-512 commented on presentation/onboarding/src/main/java/com/hilingual/presentation/onboarding/component/TermsBottomSheet.kt:
```diff
@@ -33,6 +34,39 @@ import com.hilingual.core.designsystem.component.bottomsheet.HilingualBasicBotto
+
+private enum class TermType {
+    REQUIRED,
+    MARKETING
```
p2
다른 TermType 네이밍이 `REQUIRED`라면, 얘도 `MARKETING` 보다는 `OPTIONAL` 이런 느낌이 낫지 않을까용??
필수 약관과 선택 약관을 구분하는 용으로 생각해주신 거라면요..!

nahy-512 replied:
저라면 TermType을 따로 두고 type 프로퍼티로 갖는 enum class를 따로 만들 것 같기도.. SERVICE_TERM, PRIVACY_POLICY, MARKETING으로?? sealed class를 쓸 수도 있을 것 같아요.
근데 이건 취향 차이일 수 있겠다는 생각이 들어서 어렵네요..

angryPodo replied:
마케팅 동의 여부를 서버로 안보내면 옵셔널로 할만한데 보내야해서 저는 지금이 낫다고 봅니다. 추후에 선택동의지만 서버에 보내지 않는다면? 옵셔녈에서도 한번 더 걸러야겠죠

angryPodo committed:
Merge branch 'develop' into feat/#215-implement-terms-agreement-bottom-sheet

nhyeonii commented on core/designsystem/src/main/java/com/hilingual/core/designsystem/component/bottomsheet/HilingualBasicBottomSheet.kt:
```diff
@@ -31,7 +31,9 @@ import com.hilingual.core.designsystem.theme.HilingualTheme
     modifier: Modifier = Modifier,
-    sheetState: SheetState = rememberModalBottomSheetState(),
+    sheetState: SheetState = rememberModalBottomSheetState(
+        skipPartiallyExpanded = true
```
p3 : 이 skipPartiallyExpanded 를 검색해 보니까 BottomSheet가 중간(절반)까지만 열리는 상태를 건너뛰게 하는 옵션임을 알게되었습니다. 혹시 이렇게 설정하신 이유가 UX 단순화를 위함일까욧 ?? 의도적으로 바로 닫힘/완전 열림만 사용하려고 하신 게 맞을까용 ?!

angryPodo replied:
@nhyeonii 화면 비율을 바꿀 경우의 기기대응 문제로 인해서 모든 컨텐츠 표시하도록 변경 했습니다ㅎㅎ
